src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_parquet.rs

```diff
@@ -128,6 +129,15 @@ impl InputFormat for InputFormatParquet {
         Ok(infos)
     }
 
+    async fn infer_schema(&self, path: &str, op: &Operator) -> Result<DataSchema> {
+        let obj = op.object(path);
+        let size = obj.content_length().await?;
+        let mut reader = obj.seekable_reader(..size);
```

Comment:
Need @Xuanwo review  here

Reply:
parquet need seek to read meta at file end.

Reply:
@youngsofun A small tips, we don't need to get the size first, we can call `obj.seekable_reader(..)` directly.